assembly/src/parsers/tests.rs

```diff
@@ -381,7 +381,7 @@ fn test_unterminated_proc() {
 
     let result = parse_module(source);
     match result {
-        Ok(_) => assert!(false),
```

Comment:
If you compile with optimizations turned on, the compiler is allowed to remove asserts.  So changing this to a panic makes sure that our tests are reliable for the optimized build, too.  (Which is important, because that's most likely the build that users will use, so we better test it.)